test/cppsuite/tests/cursor_bound_03.cpp

```diff
@@ -0,0 +1,423 @@
+                    }
+                } else {
+                    tc->txn.rollback();
+                    ++rollback_retries;
+                }
+                testutil_assert(rollback_retries < MAX_ROLLBACKS);
+
+                /* Sleep the duration defined by the configuration. */
+                tc->sleep();
+            }
+
+            /* Rollback any transaction that could not commit before the end of the test. */
+            tc->txn.try_rollback();
+
+            /* Reset our cursor to avoid pinning content. */
+            testutil_check(cc.cursor->reset(cc.cursor.get()));
+            if (++counter == ccv.size())
+                counter = 0;
+            testutil_assert(counter < collections_per_thread);
+        }
+    }
+
+    void
+    read_operation(thread_worker *tc) override final
+    {
+        /*
+         * Each read operation performs search_near calls with and without prefix enabled on random
+         * collections. Each prefix is randomly generated. The result of the search_near call with
+         * prefix enabled is then validated using the search_near call without prefix enabled.
+         */
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        const char *key_prefix;
+        int exact_prefix, ret;
+        int64_t prefix_size;
+        std::map<uint64_t, scoped_cursor> cursors;
+        std::string generated_prefix, key_prefix_str;
+
+        while (tc->running()) {
+            /* Get a random collection to work on. */
+            collection &coll = tc->db.get_random_collection();
+
+            /* Find a cached cursor or create one if none exists. */
+            if (cursors.find(coll.id) == cursors.end())
+                cursors.emplace(coll.id, std::move(tc->session.open_scoped_cursor(coll.name)));
+
+            auto &cursor_prefix = cursors[coll.id];
+
+            wt_timestamp_t ts = tc->tsm->get_valid_read_ts();
+            /*
+             * The oldest timestamp might move ahead and the reading timestamp might become invalid.
+             * To tackle this issue, we round the timestamp to the oldest timestamp value.
+             */
+            tc->txn.begin(
+              "roundup_timestamps=(read=true),read_timestamp=" + tc->tsm->decimal_to_hex(ts));
+
+            while (tc->txn.active() && tc->running()) {
+                /*
+                 * Generate a random prefix. For this, we start by generating a random size and then
+                 * its value.
+                 */
+                prefix_size = random_generator::instance().generate_integer(
+                  static_cast<int64_t>(1), tc->key_size);
+                generated_prefix = random_generator::instance().generate_random_string(
+                  prefix_size, characters_type::ALPHABET);
+
+                /* Call search near with the prefix cursor. */
+                bound_set prefix_bound = bound_set(generated_prefix.c_str());
```

Comment:
Here as well.

Reply:
Do we need to pass in the C style string into the constructor, can we just use generated_prefix instead?

Reply:
[461c66e](https://github.com/wiredtiger/wiredtiger/pull/8185/commits/461c66e1f16525f1a84449da15c67e4a664ef2e4)